src/odemis/gui/cont/tabs/cryo_chamber_tab.py

```
@@ -373,6 +375,16 @@ def _change_project_conf(self, new_dir):
         self.conf.pj_ptn, self.conf.pj_count = guess_pattern(new_dir)
         self.txt_projectpath.Value = os.path.basename(self.conf.pj_last_path)
         self.tab_data_model.main.project_path.value = new_dir
+        # Decide once per project whether features created during this session are
+        # eligible for data collection. Stored as a dynamic attribute — not part
+        # of the formal model — and read by add_new_feature via getattr.
+        probability = get_data_collector().probability
+        self.tab_data_model.main.collect_features = (random.random() < probability)
```

Ruff (0.16.2)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)